### PR TITLE
Update module node-github to 12.1.0

### PR DESCRIPTION
**Pull request by cosminghitea**
Hello, I update this tool to use the node-github module 12.1.0 also I add a flag for the proxy. I test the tool manually but I didn't test the test you made.

### CODE REVIEW

**Comment by bchr02** [comment]
@cosminghitea Thank you for this PR!

To run the tests, install the dev dependencies and then run ```npm test```

I ran the tests and they are all failing. I believe this has to do because of the updated node-github. Therefore the mocks within test.js need to be updated. Can you make these changes and then add additional tests so that we can keep 100% coverage? Thank you!

**Comment by cosminghitea** [comment]
Sure I will take a look on that.

**Comment by cosminghitea** [comment]
Ok I repair the tests should work now

**Comment by ptusch** [comment]
I'd love to have this as well. 👍

**Comment by nox-do** [comment]
Cool. Didn't know this exists

File: index.js [hunk 1]
```diff
@@ -88,20 +88,24 @@ NodePreGypGithub.prototype.createRelease = function(args, callback) {
...
 	
 	this.github.authenticate(this.authenticate_settings());
-	this.github.releases.createRelease(options, callback);
+	this.github.repos.createRelease(options, callback);
 };
 
 NodePreGypGithub.prototype.uploadAsset = function(cfg){
 	this.github.authenticate(this.authenticate_settings());
-	this.github.releases.uploadAsset({
+	if (this.release.id === undefined && this.release.data === undefined) {
```
Comment (bchr02) [inline index.js]: Why is this necessary? Where does release.data come from? Why now default the release id?

Comment (cosminghitea) [reply index.js]: In the new version the data that I receive from response it come into an object .data(in case there are multiple information to give) or just the single object. The release.id is necessary if the member is not defined you give a default value. If you have a suggestion I be glad to look into it and change it.

**Comment by bchr02** [comment]
node-github was deprecated
